contents/flood_fill/code/csharp/FloodFill.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System.Linq;
+using System.Diagnostics;
+using System.Collections.Generic;
+
+
+namespace Graphics
+{
+    abstract class FloodFill
+    {
+        // A simple point on the 2 dimensional integer plane.
+        private class Point2I
+        {
+            // Coordinates
+            public int x;
+            public int y;
+
+            public Point2I(int x, int y)
+            {
+                this.x = x;
+                this.y = y;
+            }
+
+            public override bool Equals(object o)
+            {
+                if(!(o is Point2I other))
+                    return false;
+                return this.x.Equals(other.x) && this.y.Equals(other.y);
+            }
+
+            public override int GetHashCode()
+            {
+                int hash = 17;
+                hash = 31 * hash + x;
+                hash = 31 * hash + y;
+                return hash;
+            }
+        }
+
+        // Utility object for comparing List<List<float>> objects.
+        private class FloatListEqualityComparer : IEqualityComparer<List<float>>
+        {
+            public bool Equals(List<float> one, List<float> two)
+            {
+                if (ReferenceEquals(one, two))
+                    return true;
+                if (one == null || two == null)
+                    return false;
+                return one.SequenceEqual(two);
+            }
```

Comment:
Couldn't you shorten this function like so?
```suggestion
            public bool Equals(List<float> one, List<float> two)
            {
                return ReferenceEquals(one, two) && one != null && two != null && one.SequenceEqual(two);
            }
```

Reply:
Let's say `one` and `two` are not `null`.  So the expression `one != null && two != null` evaluates to `true`. This way, 
`ReferenceEquals(one, two) && one != null && two != null && one.SequenceEqual(two)` can be simplified to 
`ReferenceEquals(one, two) && one.SequenceEqual(two)`. If `one` and `two` point to different `List`s, then the first argument will evaluate to false. That means that, no matter what `one.SequenceEqual(two)` (if they have equal contents or not) evaluates to, the result will always be false, which ultimately defeats the purpose of overriding of the `bool Equals(List<float>, List<float>)` function.

Reply:
So, I'm afraid the initial function and the suggested changes are not equivalent.

Reply:
It was too late, I used the wrong operator.
I meant to use `||` instead of the first `&&`

Reply:
Ah, yes that's better. It didn't cross my mind to correct your typo, I'm sorry. 
`&&` has higher priority than `||` so I believe you are right.